ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java

```diff
@@ -655,6 +658,16 @@ public Void visitFunctionCall(final FunctionCall functionCall, final Void contex
     }
 
     public void validate() {
+
+      if (containsUserColumnsWithSameNameAsNewPseudoColumns(analysis, ksqlConfig)) {
```

Comment:
Let's move this into PushQueryValidator and PullQueryValidator instead, since it's a form of query validation. Once we've done that, let's add unit test coverage in the relevant files.